clients/client-core/src/client/real_messages_control/real_traffic_stream.rs

```diff
@@ -409,28 +633,47 @@ where
                 next_delay.as_mut().reset(next_poisson_delay);
             }
 
-            // check if we have anything immediately available
-            if let Some(real_available) = self.received_buffer.pop_front() {
-                return Poll::Ready(Some(StreamMessage::Real(Box::new(real_available))));
+            // In addition to closing connections on receiving such messages, also close
+            // connections when sufficiently stale.
+            self.received_buffer.prune_stale_connections();
```

Comment:
while I don't think it's a huge problem in this case, we generally want to avoid doing a lot of work in `poll` method